packages/nimble-components/src/table/specs/table-row-grouping-hld.md

```diff
@@ -0,0 +1,228 @@
+
+A naive implementation for the custom element to use for rendering group header values for the `TableColumnText` might look like the following:
+
+```
+export class TableColumnTextGroupHeaderView extends TableGroupHeaderView<
+TableColumnTextColumnConfig
+> {
+    @observable
+    public override columnConfig!: TableColumnTextColumnConfig;
+
+    @volatile
+    public get content(): string {
+        return typeof this.groupHeaderValue === 'string'
+            ? this.groupHeaderValue
+            : this.columnConfig.placeholder;
+    }
+}
+
+const tableColumnTextGroupHeaderView = TableColumnTextGroupHeaderView.compose({
+    baseName: 'table-column-text-group-header',
+    template: html<TableColumnTextGroupHeaderView>`
+        <nimble-text-field readonly="true" value="${x => x.groupHeaderValue}" placeholder="${x => x.columnConfig.placeholder}">
+        </nimble-text-field>`,
+    styles: /* styling */
+});
+DesignSystem.getOrCreate().withPrefix('nimble').register(tableColumnTextGroupHeaderView());
+```
+
+### Rendering the row group
+
+Rendering a row group has concerns beyond just rendering the grouping value. It also has a button for controlling the expand/collapse state of a row group (indented for sub-groups) as well as potentially rendering auxiliary information such as the total number of items within a group. We can provide a new `TableGroupRow` element to manage all of these:
+
+```
+export interface TableGroupRowState {
+    rowGroupValue: unknown;
+    level: number;
+    leafItemCount: number;
+    columnConfig: unknown;
+    isExpanded: boolean;
+    rowGroupHeaderViewTag: string;
+}
+
+export class TableGroupRow extends FoundationElement {
+    @observable
+    public rowGroupState: TableGroupRowState;
+
+    public onExpandRowToggle(): void {
+        const detail: TableGroupRowExpandToggleEventDetail = {
+        };
+        this.$emit('on-row-group-expand-toggle', detail);
+    }
+
+    ...
+}
+
+const tableGroupRowElement = TableGroupRow.compose({
+    baseName: 'table-row-group',
+    template: html<TableColumnTextGroupHeaderView>`
+    <template
+        role="rowgroup"
+        style="--ni-private-row-group-indent-level: ${x => x.rowGroupState.level};"
+    >
+        <nimble-button
+            appearance="ghost"
+            content-hidden
+            @click=${x => x.onExpandRowToggle()}>
+            ${when(x => x.rowGroupState.isExpanded, html`
+                <nimble-icon-arrow-expander-down slot="start"></nimble-icon-arrow-expander-down>
+            `)}
+            ${when(x => !x.rowGroupState.isExpanded, html`
+                <nimble-icon-arrow-expander-right slot="start"></nimble-icon-arrow-expander-right>
+            `)}
+            Expand/Collapse
```

Comment:
I believe the right answer to this is to actually remove this altogether. We don't visualize that content at all, nor should we, as the `TreeView` does not do this. Ultimately, I believe the visual design for the expand/collapse button here should match what is in place for a `TreeItem`. 

I believe we may need some more specific guidance from @NIbokeefe or @RickA-NI to cover:
1. What the mouseover states are for the expand/collapse icon for a group row (for the `TreeItem` there aren't any)?
2. What is the expected cursor behavior when hovering over the header visual (typically the text) for a group row (should the cursor be an I-beam, etc..)?

I will add these questions to the 'Open Questions' section.

Reply:
@NIbokeefe @leslieab feel free to chime in if you disagree.

1) The icon itself shouldn't have any. If we add anything it would be on the group row as a whole. We've been referencing what we do on our tree control for this grouping stuff. We should follow the pattern there and have the same rollover and pressed states as the rows (I realize the default background color difference may require some tweaking). In the tree control it appears like the expand/collapse is limited to the actual icon itself and we allow the group row to be selected. Is that needed? I would think that a click anywhere in the group row should expand or collapse the section not just the tiny area of the icon.
2) The I-beam cursor implies the ability to select and/or edit the text. Do we need to allow that? The reason I ask is because if we have the behavior I described above where a click anywhere in the row expands and collapses the section then the default cursor would be preferrable and we probably wouldn't be able to select the text.

Reply:
@RickA-NI It is worth noting that:
- Group rows can be both expanded/collapsed and selected (to select/indicate selection on all data rows in the group)
- We will eventually have hierarchical data rows, which can be both expanded/collapsed and selected

The primary click action on data rows is selection. I am fine with making the primary click action on group rows expand/collapse. However, the primary click action on hierarchical data rows should be selection and not expand/collapse. In the last case, it is likely useful for the expand/collapse button to have a hover state in order to indicate that the user is hovering over a hot spot that does something other than the primary click action. Having an interaction model where the user clicks a group row to expand/collapse and the group row checkbox to select but clicks the hierarchical data row to select and the hierarchical data row expand/collapse button to expand/collapse may be more prone to user error than keeping the interactions consistent for both.

Reply:
I've added a 'Row interaction summary' section that encapsulates the expected visual behaviors of the group rows.